source/math/Quaternion.ooc

```diff
@@ -260,6 +262,80 @@ Quaternion: cover {
 		}
 		result
 	}
+	weightedMeanOfQuaternions: static func (quaternions: VectorList<This>, weights: Float[]) -> This {
+		// Implementation of the QUEST algorithm. For variable meanings, see original publication:
+		// M.D. Shuster and S.D. Oh, "Three-Axis Attitude Determination from Vector Observations", 1981
+		// [http://www.malcolmdshuster.com/Pub_1981a_J_TRIAD-QUEST_scan.pdf]
+		// TODO: Fix singularity problem when the angle is close to PI, using sequential rotations
+		vectorCount := 3 * quaternions count
+		referenceVectors := FloatMatrix new(vectorCount, 3)
+		observationVectors := FloatMatrix new(vectorCount, 3)
+		This _createVectorMeasurementsForQuest(quaternions, referenceVectors, observationVectors)
+		This _normalizeFloatArray(weights, quaternions count)
+
+		B := FloatMatrix new(3, 3)
```

Comment:
If you need to read some paper to understand what's going on here, it has benefits. Personally, I would prefer more descriptive names... @simonmika ?